llvm/lib/CodeGen/GlobalMergeFunctions.cpp

```diff
@@ -666,7 +666,6 @@ bool GlobalMergeFuncPassWrapper::runOnModule(Module &M) {
 
 PreservedAnalyses GlobalMergeFuncPass::run(Module &M,
                                            AnalysisManager<Module> &AM) {
-  ModuleSummaryIndex *Index = &(AM.getResult<ModuleSummaryIndexAnalysis>(M));
-  bool Changed = GlobalMergeFunc(Index).run(M);
+  bool Changed = GlobalMergeFunc(ImportSummary).run(M);
```

Comment:
If GlobalMergeFunc doesn't need ModuleSummaryIndex, why not just initialize without the parameter? Is there some weird side effect from some other thing that requires passing it in even though it's not used?

Reply:
This pass only uses this combined summary to determine if it's a LTO mode -- https://github.com/llvm/llvm-project/blob/main/llvm/lib/CodeGen/GlobalMergeFunctions.cpp#L539-L542.

Currently, this pass is utilized in a pre-codegen pass that employs the legacy `createGlobalMergeFuncPass()`. As per the review comment, I have also incorporated an option for the new pass manager, `GlobalMergeFuncPass()`. If we intend to use this with the typical new PM, such as `buildThinLTODefaultPipeline()`, it is necessary to pass `ImportSummary` to the constructor.